### ISSUE TITLE
Output memmap pattern

### ISSUE THREAD

**Issue by multimeric** [memtest86plus/memtest86plus#288]
Along with BadRAM patterns, another possible way to provide a "mask" that disables non-functional RAM is using a [memmap `!` flag](https://www.kernel.org/doc/html/v4.14/admin-guide/kernel-parameters.html#:~:text=memmap%3Dnn%5BKMG%5D!ss%5BKMG%5D). I would like to suggest as a feature that `memtest86+` allows the output of a single memmap pattern.

There are two immediate advantages of using memmap over BadRAM patterns that I can think of:

1. `memmap` is a Linux kernel flag, whereas BadRAM is either a [GRUB option](https://www.gnu.org/software/grub/manual/grub/html_node/badram.html) (and GRUB isn't guaranteed to be used, e.g. Pop!_OS uses systemd-boot) or [a kernel patch](https://help.ubuntu.com/community/BadRAM). So a `memmap` flag is much more likely to be useable by the average Linux user
2. Copying BadRAM codes is very difficult because they are only printed to the screen and not saved to disk. e.g. in my case taking a photo of the screen and writing all these addresses "manually" without making any mistakes is very tricky: 
![image](https://user-images.githubusercontent.com/5019367/227755378-eb778a56-4166-42f3-b861-85a049e30df1.png)

In contrast to the above, the memmap pattern that I ultimately used was `memmap=1M!2399M` which is extremely easy to write out even from an image.

**Comment by x86fr** [comment]
Fully agree. That would be a nice addition.

**Comment by vt-alt** [comment]
Yes, readme references `Linux BadRAM feature` but there is no such feature (it ever was?) It is external patch, abandoned around 2008, and AFAIK no distribution kernel supports it.

But `GRUB_BADRAM` and `memmap` are actually present features. So printing in their format would more useful.

**Comment by DAM1173** [comment]
I also miss the memmap mode in Memtest86+ v7.20, in the 'Error reporting Mode' I only see:
```
<F1> Error Counts only
<F2> Error summary
<F3> Individual errors
<F4> BadRAM patterns
```

While at https://memtest.org/readme under 'Error Reporting' the memmap mode is indeed mentioned:

> Linux memmap
> 
> The Linux memmap mode accumulates and displays faulty memory regions for use with the [Linux memmap boot command line option] (https://www.kernel.org/doc/Documentation/admin-guide/kernel-parameters.txt). Lines are printed in the form memmap=S1$A1,S2,A2... In each S,A pair, the A represents the first address in the region and the S is the size of the region (in bytes). Up to 20 faulty memory regions are recorded. Once more than 20 regions of contiguous faulty locations have been found, regions will be merged, which will mean some regions include non-faulty locations. The program will try to minimise the number of non-faulty locations that are included.

How do I enable the memmap mode under 'Error Reporting Mode'? Otherwise you have to convert it from 'BadRAM patterns' to 'memmap=' manually, see https://askubuntu.com/questions/1435576/how-to-block-ubuntu-from-using-certain-memory-area "[How to block Ubuntu from using certain memory area?](https://askubuntu.com/questions/1435576/how-to-block-ubuntu-from-using-certain-memory-area)".

**Comment by martinwhitaker** [comment]
Support for memmap mode was added in PR #454 and will be present in the next release. If you need it now, you can use one of the nightly builds.
